src/libpriv/rpmostree-postprocess.c

```diff
@@ -1525,7 +1525,12 @@ rpmostree_treefile_postprocessing (int            rootfs_fd,
       g_autofree char *dest_default_target_path =
         g_strconcat ("/usr/lib/systemd/system/", default_target, NULL);
 
-      static const char default_target_path[] = "usr/etc/systemd/system/default.target";
+      /* This used to be in /etc, but doing it in /usr makes more sense, as it's
+       * part of the OS defaults. This was changed in particular to work with
+       * ConditionFirstBoot= which runs `systemctl preset-all`:
+       * https://github.com/projectatomic/rpm-ostree/pull/1425
```

Comment:
Hi, I think I understand the parts for `ConditionFirstBoot`, but changing the target path to `/usr/lib/` seems a bit confusing to me.

I think i am lacking a bit of context, but out of curiosity, i am going to have a guess for the reasoning here :). I am thinking `default.target` could be changed by `systemctl preset`, and if `default.target` remains at `/usr/etc`, it will be counted as `modified file` during the 3 way merge process? And to avoid that, we move it to /usr/lib? 

P.S: if my guess is wrong, would you mind elaborating the reasoning a bit more?  Thanks.

Reply:
No problem!  It's actually unrelated to ostree; this is about what happens the first boot.  See this bit in systemd:
```
        /* If this is the first boot, and we are in the host system, then preset everything */
        if (m->first_boot > 0 &&
            MANAGER_IS_SYSTEM(m) &&
            !m->test_run_flags) {

                r = unit_file_preset_all(UNIT_FILE_SYSTEM, 0, NULL, UNIT_FILE_PRESET_ENABLE_ONLY, NULL, 0);
                if (r < 0)
                        log_full_errno(r == -EEXIST ? LOG_NOTICE : LOG_WARNING, r,
                                       "Failed to populate /etc with preset unit settings, ignoring: %m");
                else
                        log_info("Populated /etc with preset unit settings.");
        }

```

That resets *all* state in `/etc/systemd/system` which includes changes that rpm-ostree wrote.  Moving it to `/usr` avoids that.

Reply:
Ah, I see, makes sense. Thanks :).